day_10/task_2/tests/pipe_maze_tests.cpp

```diff
@@ -129,6 +129,45 @@ std::vector<MazeScopeParams> GenerateMazeScopeParams()
     };
 }
 
+INSTANTIATE_TEST_SUITE_P(IsInMazeScope,
+                         MazeScopeTests,
+                         testing::ValuesIn(GenerateMazeScopeParams()));
+
+
+TEST_P(MazeScopeTests, GivenMazePoint_WhenColorNeighborForPoint_ExpectColorEightPoint)
+{
+    // const PipeMaze pipe_maze = {
+    // "- L | F 7",
+    // "7 S - 7 |",
+    // "L | 7 | |",
+    //  "- L - J |",
+    //  "L | -J F"};
+
+    auto [color_right, color_top_right, color_down_right, color_top, color_down,
+          color_left, color_top_left, color_down_left] = GetParam();
+
+    maze.color_neighbor(MazePoint{1, 1});
+    ASSERT_EQ(check_state_at(MazePoint{1, 2}), color_right);
+    ASSERT_EQ(check_state_at(MazePoint{0, 2}), color_top_right);
+    ASSERT_EQ(check_state_at(MazePoint{2, 2}), color_down_right);
+
+    ASSERT_EQ(check_state_at(MazePoint{0, 1}), color_top);
+    ASSERT_EQ(check_state_at(MazePoint{2, 1}), color_down);
+
+    ASSERT_EQ(check_state_at(MazePoint{0, 0}), color_top_left);
+    ASSERT_EQ(check_state_at(MazePoint{1, 0}), color_left);
+    ASSERT_EQ(check_state_at(MazePoint{2, 0}), color_down_left);
+}
+
+std::vector<MazeScopeParams> GenerateMazeScopeParams()
+{
+
+    return std::vector<MazeScopeParams>{
+        {State::Loop, State::Red, State::Green, State::Red, State::Loop,
+         State::Red, State::Red, State::Red},
```

Comment:
MazeScopeParams is
```cpp
using MazeScopeParams = std::tuple<MazePoint, bool>;
```

hence this vector is wrongly defined.

Reply:
done